src/Turing.jl

```diff
@@ -138,5 +138,10 @@ export  @model,                 # modelling
         elementwise_loglikelihoods,
         generated_quantities,
         logprior,
-        logjoint
+        logjoint,
+
+        get_tempered_loglikelihoods_and_params,     # Tempering 
+        make_tempered_model,
+        make_tempered_loglikelihood,
+        get_params
```

Comment:
Sorry for being annoying but it is still unclear to me why these functions have to be exported - `get_params` is an internal function and should really not become part of the official API.

Maybe it's easier to understand and discuss if you show an example of how this is used in practice? BTW one should add such examples to the tests in any case :slightly_smiling_face:

Reply:
Ah yes, that was another thing, I meant to ask, where is the `params` functions defined? If it is then yeah I can remove this altogether, and I perhaps resolved the comment and hid my own comment but I realised that yes these don't need to be exported, about to push an amended commit where they are no longer exported

Reply:
It is defined in Inference.jl.

Reply:
I see yeah, I suppose maybe my version is justified if I am not exporting it, as in general we are taking in either a state (could be `HMCState` etc.) or a Transition, and then getting theta out, this is why I defined the `get_vi` helper as some states *are* the VarInfo whilst others contain it, then I just figured the sampler was the way to access the theta, but you are suggesting I instead should just call `getparams`?

Reply:
As long as it does not duplicate some existing functionality and it is internal, I am fine with it :slightly_smiling_face:

Reply:
Actually though, `getparams` returns a named tuple type thing, whereas I just want the ordered theta, so I think indexing by `sampler` is the only way to achieve this